Add smart pointer to RenderScrollbarPart

abigailfox opened the pull request:
#### 1ec83c596c2aef29e45dedabbc8358bea6fb6155
<pre>
Add smart pointer to RenderScrollbarPart
<a href="https://bugs.webkit.org/show_bug.cgi?id=280740">https://bugs.webkit.org/show_bug.cgi?id=280740</a>

Reviewed by Wenson Hsieh.

Add WeakPtr to RenderScrollbarPart.h to address static analyzer errors in both RenderScrollbarPart.h and RenderScrollbarPart.cpp

* Source/WebCore/rendering/RenderScrollbarPart.h:

Canonical link: <a href="https://commits.webkit.org/284563@main">https://commits.webkit.org/284563@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a4df2b3a738ea04248120bd14be549862763255e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69814 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/49215 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22567 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73899 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20972 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71931 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/57015 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20823 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/55431 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13901 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72880 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44846 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60209 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35911 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41511 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17649 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19349 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63440 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17993 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75613 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/14039 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17234 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/63121 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/14074 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60290 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63050 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15505 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11055 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4669 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/45018 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/46092 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/47363 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45833 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/rendering/RenderScrollbarPart.h:
```diff
@@ -66,7 +66,7 @@ class RenderScrollbarPart final : public RenderBlock {
     void computeScrollbarHeight();
     
-    RenderScrollbar* m_scrollbar;
+    RefPtr<RenderScrollbar> m_scrollbar;
```
Hi, Abigail :).

Since `RenderScrollbar` owns `RenderScrollbarPart` (in `RenderScrollbar::m_parts`), this will create a reference cycle. We probably want to use WeakPtr<RenderScrollbar> instead.

abigailfox replied:
I'm actually making that change right now, Wenson checked me on that! (I think this is the first time I've run into a need for WeakPtr organically in the wild?)

abigailfox replied:
It looks like the base for RenderScrollbar (Widget) supports `SingleThreadWeakPtr` so that's what I've added here as regular `WeakPtr` doesn't seem to be compatible

whsieh reviewed (approved):
r=mews

abigailfox committed:
Add smart pointer to RenderScrollbarPart
https://bugs.webkit.org/show_bug.cgi?id=280740

Reviewed by Wenson Hsieh.

Add WeakPtr to RenderScrollbarPart.h to address static analyzer errors in both RenderScrollbarPart.h and RenderScrollbarPart.cpp

* Source/WebCore/rendering/RenderScrollbarPart.h:

Canonical link: https://commits.webkit.org/284563@main
